### DISCUSSION TITLE
How to migrate from ReportHTML to PSWriteHTML

### DISCUSSION THREAD

**Discussion by BrooksV** [EvotecIT/PSWriteHTML#341] [Q&A] [answered]
I am having difficulty migrating my existing scripts from ReportHTML to PSWriteHTML.

Any docs, suggestions, and ins-and-outs on how to update my scripts?

My Scripts are using most of these ReportHTML CmdLets:

Get-HTMLCSS
Get-HTMLLogos
Get-HTMLClosePage 
Get-HTMLColumnClose
Get-HTMLColumnOpen
Get-HTMLContentClose
Get-HTMLContentdatatable
Get-HTMLContentdatatable -ArrayOfObjects ( $SubscriptionRGs)
Get-HTMLContentOpen
Get-HTMLContentOpen -HeaderText "Complete List"
Get-HTMLContentTable
Get-HTMLContentText
Get-HTMLHeading
Get-HTMLOpenPage
Get-HTMLPieChartObject
Save-HTMLReport
Set-TableRowColor

**Comment by PrzemyslawKlys** [comment, marked as answer]
You can forget about needing to Close anything. The concept of saving is also taken care of by providing filePath to New-HTML, so there's no need to think about it. I recommend starting from scratch than trying to convert your existing commands to new commands because you can create your new output in much less code. 

Your commands will be:
- New-HTML (used once to define HTML content) with FilePath to save the file. Using the Online switch will tell it to use "Links" to JS/CSS instead of inline css/js code, which will be smaller in size but will require you to be online to view the page.  You tell the end of the HTML is by placing closing bracket in the end. Most of the functions work this way. 

```powershell
New-HTML {
    # your content
} 
```

1 section/column with header text. This section holds a table

```powershell
New-HTML {
    New-HTMLSection -HeaderText "Subscription Resource Groups" {
          New-HTMLTable -DataTable $SubscriptionRGS
    } 
} -FilePath .\Example7_01.html -Online -ShowHTML
```

2 sections... 1 that holds table, the other one being empty - notice that they go row by row

```powershell
New-HTML {
    New-HTMLSection -HeaderText "Subscription Resource Groups" {
        New-HTMLTable -DataTable $SubscriptionRGS
    }
    New-HTMLSection -HeaderText "your next row" {

    }
} -FilePath .\Example7_01.html -Online -ShowHTML
```
Same 2 sections - but if you nest them into New-HTMLSection they will become columns. If you add more sections within that nested section it will keep on adding columns. 

```powershell
New-HTML {
    New-HTMLSection -HeaderText "Notice how nesting this makes it work on columns" {
        New-HTMLSection -HeaderText "Subscription Resource Groups" {
            New-HTMLTable -DataTable $SubscriptionRGS
        }
        New-HTMLSection -HeaderText "your next row" {

        }
    }
} -FilePath .\Example7_01.html -Online -ShowHTML
```

If you want to hide that "top" section - just add invisible switch for the top section

```powershell
New-HTML {
    New-HTMLSection -HeaderText "Notice how nesting this makes it work on columns" {
        New-HTMLSection -HeaderText "Subscription Resource Groups" {
            New-HTMLTable -DataTable $SubscriptionRGS
        }
        New-HTMLSection -HeaderText "your next row" {

        }
    } -Invisible
} -FilePath .\Example7_01.html -Online -ShowHTML
```

If you want to add logo to it just add New-HTMLLogo inside New-HTML. 

```powershell
New-HTML {
    New-HTMLLogo
    New-HTMLSection -HeaderText "Notice how nesting this makes it work on columns" {
        New-HTMLSection -HeaderText "Subscription Resource Groups" {
            New-HTMLTable -DataTable $SubscriptionRGS
        }
        New-HTMLSection -HeaderText "your next column" {

        }
    } -Invisible
} -FilePath .\Example7_01.html -Online -ShowHTML
```

![image](https://user-images.githubusercontent.com/15063294/195421527-71f2baf6-e798-4170-a9dd-fe993b932170.png)


Charts are a bit more complicated. You define New-HTMLChart and then based on what you want to show you add Pies

```powershell
            New-HTMLChart {
                New-ChartPie -Name 'Pie Part 1' -Value 20
                New-ChartPie -Name 'Pie Part 2' -Value 30
                New-ChartPie -Name 'Pie Part 3' -Value 40
            }
```

It looks like this when used together:

```powershell
New-HTML {
    New-HTMLLogo
    New-HTMLSection -HeaderText "Notice how nesting this makes it work on columns" {
        New-HTMLSection -HeaderText "Subscription Resource Groups" {
            New-HTMLTable -DataTable $SubscriptionRGS
        }
        New-HTMLSection -HeaderText "your next row" {
            New-HTMLChart {
                New-ChartPie -Name 'Pie Part 1' -Value 20
                New-ChartPie -Name 'Pie Part 2' -Value 30
                New-ChartPie -Name 'Pie Part 3' -Value 40
            }
        }
    } -Invisible
} -FilePath .\Example7_01.html -Online -ShowHTML
```
![image](https://user-images.githubusercontent.com/15063294/195421415-5be6e687-a4c2-451e-8a98-44f7b4748b47.png)


If you make a screenshot I can maybe mock something up.

**Reply by BrooksV** [reply]
Thanks Kyls

Here are some things I am having problems figuring out:

- How to specify my CCS style sheet?
- How to get a footer line that has say a CopyRight on the left and the Generated Text on the right?

In the past I did things like:

.copyright {
 float:left;
 text-align: left;
}
.generated {
 float:right;
 text-align: right;
}

<div id="footer">
    <h3 class="copyright">Copyright ....</h3>
    <h3 class="generated">Generated: Friday, October 14, 2022 10:24:23</h3>
    <div style="clear: both"></div>
</div>

Thanks In Advance
